packages/cosmic-swingset/x/swingset/ibc.go

```diff
@@ -97,115 +77,358 @@ func (ch channelHandler) Receive(ctx *ControllerContext, str string) (ret string
 		blockTimeout := int64(keeper.DefaultPacketTimeout)
 
 		packet := channeltypes.NewPacket(
-			msg.GetData(), seq,
+			msg.Data64, seq,
```

Comment:
Can you please take these fields from `msg.Packet`?  Then we don't need `msg.Tuple` at all.